src/internal/connector/exchange/exchange_data_collection.go

```diff
@@ -139,33 +144,83 @@ func (col *Collection) populateByOptionIdentifier(
 		return
 	}
 
+	limitCh := make(chan struct{}, urlPrefetchChannelBufferSize)
+	defer close(limitCh)
```

Comment:
Could this be added to the defer on  Line 135?

Reply:
I thought it would be better to couple it with the `make` as the defer above is just for the progressbar.